File: app/src/main/java/com/boardgamegeek/provider/GamesRanksProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.GameRanks;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesRanksProvider extends BaseProvider {

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		return new SelectionBuilder().table(Tables.GAME_RANKS);
	}

	@Override
	protected String getDefaultSortOrder() {
		return GameRanks.DEFAULT_SORT;
	}

	@Override
	protected String getPath() {
		return "games/ranks";
	}

	@Override
	protected String getType(Uri uri) {
		return GameRanks.CONTENT_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesRanksProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import com.boardgamegeek.provider.BggContract.*
import com.boardgamegeek.provider.BggContract.Companion.PATH_GAMES
import com.boardgamegeek.provider.BggContract.Companion.PATH_RANKS
import com.boardgamegeek.provider.BggDatabase.Tables

class GamesRanksProvider : BaseProvider() {
    override fun getType(uri: Uri) = GameRanks.CONTENT_TYPE

    override val path = "$PATH_GAMES/$PATH_RANKS"

    override val defaultSortOrder = GameRanks.DEFAULT_SORT

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        return SelectionBuilder().table(Tables.GAME_RANKS)
    }
}
